weave/flow/monitor.py

```diff
@@ -0,0 +1,119 @@
+from pydantic import Field, field_validator
+from typing_extensions import Self
+
+from weave.flow.casting import ScorerLike
+from weave.flow.obj import Object
+from weave.trace.api import ObjectRef, publish
+from weave.trace.objectify import register_object
+from weave.trace.vals import WeaveObject
+
+
+@register_object
+class Monitor(Object):
+    """
+    Sets up a monitor to score incoming calls automatically.
+
+    Examples:
+
+    ```python
+    import weave
+    from weave.scorers import ValidJSONScorer
+
+    json_scorer = ValidJSONScorer()
+
+    my_monitor = weave.Monitor(
+        name="my-monitor",
+        description="This is a test monitor",
+        sampling_rate=0.5,
+        call_filter={
+            "op_names": ["my_op"],
+            "query": {
+                "$expr": {
+                    "$gt": [
+                        {
+                            "$getField": "started_at"
+                        },
+                        {
+                            "$literal": 1742540400
+                        }
+                    ]
+                }
+            }
+        },
+        scorers=[json_scorer],
+    )
+
+    my_monitor.activate()
+    ```
+    """
+
+    sampling_rate: float = Field(ge=0, le=1)
+    scorers: list[ScorerLike]
```

Comment:
While `ScorerLike` allows for op-based scores, I think pinning to the scorer class itself will allow for simpler code and benefits later on.

Reply:
To elaborate: the `op` style is more of a user convenience for evaluations and in practice, we are only going to allow a very specific subset of allowed scorers here anyway, which will all be vetted classes.